proxy/envoy/config.go

```diff
@@ -163,6 +180,43 @@ func build(instances []*model.ServiceInstance, services []*model.Service,
 				},
 			}},
 		}
+
+		// Add TCP filters that share a port with HTTP filters.
+		if _, ok := tcpRouteConfigs[port]; ok {
+
+			// TCP and HTTP filter on the same port seems to cause
+			// envoy to assign all traffic to TCP filter regardless of
+			// the routing config. Temporarily drop TCP filters that
+			// share a port with HTTP (e.g. istio-egress) until this is
+			// root caused.
+
```

Comment:
I see, this is a pretty important issue. So essentially, if I have service A on TCP 80 and service B on HTTP 80, A filter catches all requests to B. Does that happen if I also put A's cluster IP for A's TCP filter?